### PR TITLE
Handle NaN in BGV files when converting to JSON

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

**Comment by tomstuart** [comment]
The code looks sensible to me. How can I try it (with & without this change) to see if it’s working right?

File: bin/bgv2json [hunk 1]
```diff
@@ -3,6 +3,23 @@
 require 'seafoam'
 require 'json'
 
+def prepare_json(object)
+  case object
+  when Float
+    if object.nan?
+      'NaN'
```
Comment (tomstuart) [inline bin/bgv2json:10]: Who’s going to be consuming this JSON? Is providing a non-numeric value better than e.g. omitting the key?

File: bin/bgv2json [hunk 2]
```diff
@@ -3,6 +3,23 @@
...
+    else
+      object
+    end
+  when Array
+    object.map { |o| prepare_json(o) }
+  when Hash
+    object.map { |k, v| [k, prepare_json(v)] }.to_h
```
Comment (tomstuart) [inline bin/bgv2json:17]:
```suggestion
    object.transform_values { |v| prepare_json(v) }
```

File: bin/bgv2json (hunk 1 above)
Comment (chrisseaton) [reply bin/bgv2json:10]: I guess you're really saying I should add a test! I refactored in #17 to make that easier to do, and will rebase on top of that when it's merged.

**Comment by chrisseaton** [comment]
I added a spec, and an example file that now includes a NaN, `examples/ruby-fib.bgv`. So you can see the difference by running the spec, or by `bgv2json examples/ruby-fib.bgv`, which would not work before due to NaN not being allowed in JSON.

**Review by tomstuart — APPROVED** [review]
Thanks for the test and example file! This looks good to me.

I don’t know what the JSON is used for, but I assume it’s consumed by something downstream which is happy to treat “`NaN`” as a text label (perhaps to draw a graph?) and doesn’t care what it represented in the `.bgv` file.

**Comment by chrisseaton** [comment]
Some other researchers do their own analysis and were happier working with a JSON dump than they were writing a Ruby script. I also just `grep` the JSON output for a quick way to see if something is in the file or not.

**Pull request merged**